src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -1481,6 +1482,16 @@ def test_following_preceding():
     assert multiply1.following() == [c_ref, d_ref]
     assert multiply1.preceding() == [assign1, a_ref, multiply2, b_ref]
 
+    # 1d: Immediately following
```

Comment:
Please could you extend slightly just to cover the case where `immediately_follows` and `immediately_precedes` return False.

Reply:
Ah yes good point. See ec05c0403c40616d7d0d4390933a53a26bad7a15.